### PR TITLE
Improve HTML semantics and a11y of language reference

### PR DESCRIPTION
**Pull request by paulespinosa**
The language reference's HTML has been updated to be more semantically correct.
This also helps to improve the document's accessibility concerns.

Results are at https://paulespinosa.github.io/zig-langref/ for review

- Document structure has single h1, other header sections start at h2, nav sections w/ aria labels, main section
- Zig's homepage is linked, Zig Standard Library section link to it
- Tables have caption and scoping rows and columns
- Code blocks are figures with figure captions citing source files
- Change line height 1.5 to include table of contents as well
- Luminosity contrast ratios have been adjusted to 7:1
- Links have default browser underline with hover and focus effects
- Asides, definition lists, keyboard inputs, program outputs are represented semantically

Tools used to check:
WAVE plugin https://wave.webaim.org/
Firefox Accessibility Developer Tool
Lighthouse Accessibility Tool

### CODE REVIEW

**Comment by jedisct1** [comment]
This is great!

Accessibility is important, so thanks a lot for these improvements.

**Comment by andrewrk** [comment]
Wow this looks beautiful! The dark mode, especially is a brilliant improvement.

I do have a couple questions:

1. Is this bright orange, in light mode, intentional?
image


2. I noticed that page up/page down stopped working. Is that working as designed?

**Comment by AssortedFantasy** [comment]
These changes make the language reference laggy on my tablet for some reason.
(Samsung Galaxy tab S2 8.0, using chrome from the google play store).

I'm not sure why since I'm not a web developer. But I can tell that there is substantially more frame hitching and scrolling isn't as smooth.

Here is a link to some website that seems to agree with what I've seen.

Current on ziglang.org 👇
https://webpagetest.org/breakdownTimeline.php?test=210825_BiDcQX_d145abaca40ece65f7ea668b9b916bd0&run=1

This pull request. 👇
https://webpagetest.org/breakdownTimeline.php?test=210825_BiDc1F_afc271f69975582827582020e194c461&run=1

**Comment by paulespinosa** [comment]
@AssortedFantasy 
> Here is a link to some website that seems to agree with what I've seen.
> 
> Current on ziglang.org point_down
> https://webpagetest.org/breakdownTimeline.php?test=210825_BiDcQX_d145abaca40ece65f7ea668b9b916bd0&run=1
> 
> This pull request. point_down
> https://webpagetest.org/breakdownTimeline.php?test=210825_BiDc1F_afc271f69975582827582020e194c461&run=1
> 
> If you could somehow improve the perf I'd really appreciate it!

Thank you for checking! The test you provided that was done against current ziglang.org was using a different configuration than the test against this pull request. Here a link to the test against current ziglang.org using the same configuration as the test against this pull request.
https://webpagetest.org/breakdownTimeline.php?test=210825_BiDc8Q_0e45192f8acdd618863d9ebcc300c01a&run=1

I'm looking into performance issues. If anyone out there can help performance improvements, let me know.

**Comment by AssortedFantasy** [comment]
@paulespinosa 
Whoops! My bad, I didn't realize that the configuration had changed on me.

I will mention my tablet (and chrome on android) is kind of heaping garbage so it does lag a bit even on the normal language reference anyway... so these changes aren't really much of a regression even if you aren't able to improve the performance. Besides its not like its unusable, just a bit low fps.

**Comment by paulespinosa** [comment]
@andrewrk 
> 
> I do have a couple questions:
> 
>     1. Is this bright orange, in light mode, intentional?
>     2. I noticed that page up/page down stopped working. Is that working as designed?
>     3. the scrollbar hits the bottom too early:

Thank you for helping with testing it out!

1. The background colors for the Zig code and Shell were taken from the ziglang.org website. (I think of the orange color as "zig gold".)   
pic-selected-210825-0956-00

The intention is to have a way of distinguishing different kinds of source code. The source code blocks are color coded using the `#f7a41d` for Zig, `#a8b9cc` for C, and `#365d95` for JavaScript. However, as I think about it now, color coding doesn't work for everyone. Personally, I do think of the bright orange, content wide, background color as a bit distracting or overwhelming.  I'll try something a bit more subtle.

pic-selected-210825-0957-05
pic-selected-210825-0957-26

2. All keyboard navigation is supposed to work. I checked Page Up and Page Down on my end and they were functional. If they're not working, I think it might be related to the fix for point number 3.
3. I've seen this happen to me before, but I can't reproduce it again. I do notice significant flickering or sometimes disappearance of the left-hand navigation section when paging down very fast. I'm looking into these issues.

**Comment by marler8997** [comment]
this is almost literally bikeshedding but if it means anything I did find the bright orange a bit hard on the eyes

**Comment by paulespinosa** [comment]
@marler8997 Issues like eye strain are helpful to know about. I lightened the original color. Does it reduce eye strain?

File: doc/docgen.zig [hunk 1]
```diff
@@ -305,6 +305,18 @@ const Link = struct {
     token: Token,
 };
 
+const SyntaxBlock = struct {
+    source_type: SourceType,
+    name: []const u8,
+    source_token: Token,
+
+    const SourceType = union(enum) {
```
Comment (luehmann) [inline doc/docgen.zig:313]: couldn't this be just an enum instead?

Comment (paulespinosa) [reply doc/docgen.zig:313]: Yes

**Review by andrewrk — APPROVED** [review]
Alright I did a final review of the changes and ready to merge! Thank you for your patience on this one. If you don't mind doing one last conflict resolution / rebase, I'll slap that green button.

**Comment by paulespinosa** [comment]
Thank you @andrewrk

**Pull request merged**